### ISSUE TITLE
Elasticsearch sessions2 files grows fast

### ISSUE THREAD

**Issue by alekdimitrov** [StamusNetworks/SELKS#302]
Hello. This is my first encounter with IDS. I have the following setup - Proxmox for virtualizer, pfsense with haproxy for firewall, several Linux servers and several Windows servers behind. I set up and deploy SELKS, most things look fine. 
I read the logrotate tutorial and set things up like this:

logrotate elasticsearch

#!/bin/bash

/usr/bin/curator_cli delete_indices --filter_list '
[
  {
    "filtertype": "age",
    "source": "creation_date",
    "direction": "older",
    "unit": "days",
    "unit_count": 1
  },
  {
    "filtertype": "pattern",
    "kind": "prefix",
    "value": "logstash*"
  }
]
'

logrotate suricata

/var/log/suricata/eve.json {
        daily
        rotate 2
        olddir /var/log/suricata/StatsByDate/
        compress
        missingok
        notifempty
        dateext
        postrotate
            /bin/kill -HUP $(cat /var/run/suricata.pid)
        endscript
}
But i cant handle files in elasticsearch from sessions2_template. Can someone give a litle help how to rotate this elasticsearch sessions2 files. I was try to made whole new ilm policy but i have Index lifecycle errors related with point alias to write index 
For now I need only base IDS and I want to say that dont understand very well need from elasticksearch. Every litlle help that you can give me wiil be priceless for me.

Im using last elasticsearch 7.12

**Comment by pevma** [comment]
Just a quick check - did you do the initial setup  steps and they completed ok? 
https://github.com/StamusNetworks/SELKS/wiki/First-time-setup

**Comment by pevma** [comment]
What are the last lets say 50 lines in `/data/moloch/logs/viewer.log` ?

**Comment by alekdimitrov** [comment]
In short...
ERROR - Couldn't find database version.  Have you run ./db.pl host:port upgrade? TypeError: Cannot read property 'mappings' of undefined
    at /data/moloch/viewer/db.js:1173:57
    at respond (/data/moloch/node_modules/elasticsearch/src/lib/transport.js:370:9)
    at checkRespForFailure (/data/moloch/node_modules/elasticsearch/src/lib/transport.js:306:7)
    at HttpConnector.<anonymous> (/data/moloch/node_modules/elasticsearch/src/lib/connectors/http.js:173:7)
    at IncomingMessage.wrapper (/data/moloch/node_modules/lodash/lodash.js:4949:19)
    at IncomingMessage.emit (events.js:326:22)
    at endReadableNT (_stream_readable.js:1223:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)

**Comment by pevma** [comment]
Can you restart the elasticsearch service, then after that restart the molochservice please
https://github.com/StamusNetworks/SELKS/wiki/Services-status-check

**Comment by alekdimitrov** [comment]
Yes. I restart elasticsearh and after that molochservice but: 
root@ASOFTIPS:/home/asoftplus# service molochviewer status
● molochviewer.service - Moloch Viewer
   Loaded: loaded (/etc/systemd/system/molochviewer.service; disabled; vendor preset: enabled)
   Active: inactive (dead)

Apr 05 12:01:32 ASOFTIPS systemd[1]: Started Moloch Viewer.
Apr 05 12:01:40 ASOFTIPS systemd[1]: molochviewer.service: Succeeded.
Apr 09 13:36:43 ASOFTIPS systemd[1]: Started Moloch Viewer.
Apr 09 13:36:50 ASOFTIPS systemd[1]: molochviewer.service: Succeeded.

Nothing changed. Actually these sessions2 files disappeared like a smoke when I put SLEKS in production. This is very strange.

**Comment by pevma** [comment]
Is this your QA setup? 
The log basically says Es is unreachable - any customizations maybe (X-pack etc) ?

**Comment by pevma** [comment]
Anything in the end of the `/var/log/elasticsearch/elasticsearch.log` ?

**Comment by alekdimitrov** [comment]
Nope. Is not QA setup. Was preproduction setup. Now is on production.

Here is the records from elasticsearh.log
[2021-04-11T09:19:19,289][INFO ][o.e.c.m.MetadataMappingService] [ASOFTIPS] [logstash-tls-2021.04.11/yVC0zCY-RJS2-N52vBcwXQ] update_mapping [_doc]
[2021-04-11T09:19:26,313][INFO ][o.e.c.m.MetadataMappingService] [ASOFTIPS] [logstash-fileinfo-2021.04.11/YCzSuN8FQviYm_mhGkZTbA] update_mapping [_doc]
[2021-04-11T09:22:27,845][INFO ][o.e.c.m.MetadataMappingService] [ASOFTIPS] [logstash-flow-2021.04.11/fXInA1amQoOqCjXtvTJA6A] update_mapping [_doc]
[2021-04-11T09:57:37,440][INFO ][o.e.c.m.MetadataMappingService] [ASOFTIPS] [logstash-alert-2021.04.11/xo26BCqdR1-_VLdUmP12Jg] update_mapping [_doc]
[2021-04-11T13:46:46,321][INFO ][o.e.c.m.MetadataMappingService] [ASOFTIPS] [logstash-smtp-2021.04.11/-QdR1dIRT_6nNNLp93dtBw] update_mapping [_doc]
[2021-04-11T17:49:02,450][INFO ][o.e.c.m.MetadataMappingService] [ASOFTIPS] [logstash-http-2021.04.11/dTf6MoqGTW6E9aBbyckJDg] update_mapping [_doc]

**Comment by pevma** [comment]
Ok - so dashboards seem are populating fine if i understand correctly - your question is how to handle the growing `sessions2_template files` , correct   ?

**Comment by alekdimitrov** [comment]
Hello again. I think I owe an apology right now. When IDS was in the pre-production stage, session2_template files grew to 50 GB or more. The HDD was 100GB and was completely full in one day. Several times I had to delete everything in / var / log / elasticsearch, because the virtual machine stopped working and the hdd space and IDS did not work. I looked at the policies in the elastix search settings several times, but I couldn't restrict the files in any way.
Because in the next 10 days I transfer everything from a new to an old server, I had almost no time to monitor the machine, and that's why I write so rarely. It turned out that after going into production sessions2_template files stopped filling the disk, which I do not know how to explain. This was my main problem, which is why I allowed myself to open a ticket.
I am  sorry for disturbing.
